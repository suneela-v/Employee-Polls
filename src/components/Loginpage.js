import { useState } from "react";
import { connect } from "react-redux";
// import { useNavigate } from "react-router-dom";
import { setauthedUser } from "../actions/authedUser";
import { Link, useLocation, useNavigate, useParams } from "react-router-dom";

const withRouter = (Component) => {
  const ComponentWithRouterProp = (props) => {
    let location = useLocation();
    let navigate = useNavigate();
    let params = useParams();
    return <Component {...props} router={{ location, navigate, params }} />;
  };
  return ComponentWithRouterProp;
};

const Loginpage = (props) => {
  const { users } = props;
  const [username, setUsername] = useState("");
  const [pwd, setPwd] = useState("");
  const [success, setSuccess] = useState(false);
  const [error, setError] = useState(false);
  const navigate = useNavigate();
  console.log(
    props,
    props.router.location.pathname,
    "pathName",
    "loginpropssssssss"
  );
  const handleChangeForUsername = (e) => {
    setUsername(e.target.value);
  };
  const handleChangeForPwd = (e) => {
    setPwd(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    //console.log(props.locationPath, "in login submit");
    // if (props.router.location.pathname !== "/") {
    //   alert("404");
    //   navigate("/");
    //   return;
    // }
    let objForLogin = Object.entries(users).filter(
      ([key, value]) => key === username
    );

    if (objForLogin.length === 0) {
      // alert(objForLogin);
      setUsername("");
      setPwd("");
      // setError(false);
      // setSuccess(true);
      // alert(error);
      // alert(success);
      // alert("enter the details");
    }
    if (objForLogin.length && users[objForLogin[0][0]].password === pwd) {
      // setError(true);
      // setSuccess(false);
      // alert(error);
      // alert(success);
      // alert("correct credentials");
      props.dispatch(setauthedUser(username));
      let pathName = props.router.location.pathname;
      // if (pathName.includes("questions")) {
      //   console.log(props.router.location.pathname, pathName.substr(10));
      let qid = pathName.substr(11);
      let beginingQuestions = [
        "8xf0y6ziyjabvozdd253nd",
        "6ni6ok3ym7mf1p33lnez",
        "am8ehyc8byjqgar0jgpub9",
        "loxhs1bqm25b708cmbf3g",
        "vthrdm985a262al8qx3do",
        "xj352vofupe1dqz9emx13r",
      ];
      //  if (props.users[props.authedUser].answers.includes(qid)) {
      // alert(qid);
      console.log(
        props.users,
        Object.keys(users[username].answers),
        Object.keys(users[username].answers).includes(qid),
        beginingQuestions,
        beginingQuestions.includes(qid),
        qid,

        // props.users[props.authedUser],
        "newwwwwwwwwwwwwwwwwww propss in loginpage"
      );
      // }
      // }

      if (
        props.router.location.pathname === "/" ||
        props.router.location.pathname === "/add" ||
        props.router.location.pathname === "/leaderboard" ||
        // Object.keys(users[username].answers).includes(qid) ||
        beginingQuestions.includes(qid)

        // || props.users[username].answers.includes(qid)
      ) {
        // alert("in correctpath");
        let currentRoute = props.router.location.pathname;
        console.log(currentRoute, "currentRoute");
        navigate(`${currentRoute}`);
      } else {
        console.log(
          props.router.location.pathname,
          "props.router.location.pathname"
        );
        //  alert("notfound");
        navigate("/notfound");
      }
    }
    if (objForLogin.length && users[objForLogin[0][0]].password !== pwd) {
      setUsername("");
      setPwd("");
      // setError(true);
      // setSuccess(false);
      // alert(error);
      // alert("wrong credentials");
    }
  };

  return (
    <>
      <form onSubmit={handleSubmit}>
        <div className="middle">
          {/* {success && (
            <h4 data-testid="success-header">successfully logged in</h4>
          )}
          {error && <h4 data-testid="error-header1">username doesn't exist</h4>}
          */}
          <h3>Employee Polls</h3>

          <h4>Login</h4>

          <div>
            <select
              onChange={handleChangeForUsername}
              value={username}
              data-testid="username-select"
            >
              <option value="" disabled>
                Username
              </option>

              <option value="mtsamis">Mtsamis</option>
              <option value="sarahedo">sarahedo</option>
              <option value="tylermcginnis">tylermcginnis</option>
              <option value="zoshikanlu">zoshikanlu</option>
            </select>
            <select
              onChange={handleChangeForPwd}
              value={pwd}
              data-testid="input-pwd"
            >
              <option value="" disabled>
                Password
              </option>

              <option value="xyz123">xyz123</option>
              <option value="password123">password123</option>
              <option value="abc321">abc321</option>
              <option value="pass246">pass246</option>
            </select>
            {/* <input
              className="inputdesign"
              placeholder="username"
              onChange={handleChangeForUsername}
              value={username}
            /> */}
            {/* <input
              className="inputdesign"
              placeholder="password"
              onChange={handleChangeForPwd}
              value={pwd}
              data-testid="input-pwd"
            /> */}
          </div>
          <button data-testid="submit-button">Submit</button>
        </div>
      </form>
    </>
  );
};
// const mapStateToProps = ({ questions, authedUser, users }) => {
//   return { questions, authedUser, users };
// };
// export default connect(mapStateToProps)(Loginpage);

const mapStateToProps = ({ questions, authedUser, users }, props) => {
  const { id } = props.router.params;
  const { locationPath } = props.router.location.pathname;
  return { id, questions, authedUser, users, locationPath };
};
export default withRouter(connect(mapStateToProps)(Loginpage));
