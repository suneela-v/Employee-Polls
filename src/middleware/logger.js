const logger = (store) => (next) => (action) => {
  console.group(action.type);
  console.log("the action :", action);
  const returnValue = next(action);
  console.log("the sate is", store.getState());
  console.groupEnd();
  return returnValue;
};
export default logger;
