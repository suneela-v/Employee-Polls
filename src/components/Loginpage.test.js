import { render, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
//import { store } from "./app/store";
import { createStore } from "redux";
import reducer from "../reducers";
import middleware from "../middleware";
import Loginpage from "./Loginpage";
import { MemoryRouter } from "react-router";
const store = createStore(reducer, middleware);
// import NewpollContainer from "./NewpollContainer";

describe("Loginpage", () => {
  it("will verify if a string matches a regex expression passed", () => {
    var component = render(
      <MemoryRouter>
        <Provider store={store}>
          <Loginpage />
        </Provider>
      </MemoryRouter>
    );

    var usernameSelect = component.getByTestId("username-select");
    fireEvent.change(usernameSelect, { target: { value: "mtsamis" } });

    var inputpwdId = component.getByTestId("input-pwd");
    fireEvent.change(inputpwdId, { target: { value: "xyz123" } });

    var submitButton = component.getByTestId("submit-button");
    fireEvent.click(submitButton);
    expect(component.getByTestId("username-select").textContent).toMatch(/ts/);
     });
});
describe("sanpshot", () => {
  it("will verify if a string matches a regex expression passed", () => {
    var component = render(
      <MemoryRouter>
        <Provider store={store}>
          <Loginpage />
        </Provider>
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
