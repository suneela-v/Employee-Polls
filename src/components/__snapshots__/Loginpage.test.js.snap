// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sanpshot will verify if a string matches a regex expression passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="middle"
        >
          <h3>
            Employee Polls
          </h3>
          <h4>
            Login
          </h4>
          <div>
            <select
              data-testid="username-select"
            >
              <option
                disabled=""
                value=""
              >
                Username
              </option>
              <option
                value="mtsamis"
              >
                Mtsamis
              </option>
              <option
                value="sarahedo"
              >
                sarahedo
              </option>
              <option
                value="tylermcginnis"
              >
                tylermcginnis
              </option>
              <option
                value="zoshikanlu"
              >
                zoshikanlu
              </option>
            </select>
            <select
              data-testid="input-pwd"
            >
              <option
                disabled=""
                value=""
              >
                Password
              </option>
              <option
                value="xyz123"
              >
                xyz123
              </option>
              <option
                value="password123"
              >
                password123
              </option>
              <option
                value="abc321"
              >
                abc321
              </option>
              <option
                value="pass246"
              >
                pass246
              </option>
            </select>
          </div>
          <button
            data-testid="submit-button"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="middle"
      >
        <h3>
          Employee Polls
        </h3>
        <h4>
          Login
        </h4>
        <div>
          <select
            data-testid="username-select"
          >
            <option
              disabled=""
              value=""
            >
              Username
            </option>
            <option
              value="mtsamis"
            >
              Mtsamis
            </option>
            <option
              value="sarahedo"
            >
              sarahedo
            </option>
            <option
              value="tylermcginnis"
            >
              tylermcginnis
            </option>
            <option
              value="zoshikanlu"
            >
              zoshikanlu
            </option>
          </select>
          <select
            data-testid="input-pwd"
          >
            <option
              disabled=""
              value=""
            >
              Password
            </option>
            <option
              value="xyz123"
            >
              xyz123
            </option>
            <option
              value="password123"
            >
              password123
            </option>
            <option
              value="abc321"
            >
              abc321
            </option>
            <option
              value="pass246"
            >
              pass246
            </option>
          </select>
        </div>
        <button
          data-testid="submit-button"
        >
          Submit
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
